Polyhedron/demo/Polyhedron/MainWindow.cpp

```diff
@@ -2777,61 +2777,130 @@ void MainWindow::propagate_action()
 }
 
+/*
+ The two following functions allow to create files from string and strings from files.
+ This is used as a workaround of the absence of stream management in our IO system.
+ The whole to/from Base64 is used to avoid problems with binary formats. Everything is written
+ as a base64 binary string, and converted back to what it was.
+*/
+QByteArray file_to_string(const char* filename)
+{
+  std::ifstream f(filename, std::ifstream::binary);
+  // get size of file
+  f.seekg (0,f.end);
+  long size = f.tellg();
+  f.seekg (0);
+  std::ostringstream ss;
+  // allocate memory for file content
+    char* buffer = new char[size];
+
+    // read content of infile
+    f.read(buffer,size);
+
+    // write to outfile
+    ss.write(buffer,size);
+    // release dynamically-allocated memory
+    delete[] buffer;
+  //ss.write( << f.rdbuf(); // reading data
+  f.close();
+  std::string st = ss.str();
+  QByteArray ba(st.c_str(), st.size());
+  return ba;
+}
+
+void MainWindow::write_string_to_file(const QString& str, const QString &filename)
+{
+  std::ofstream f(filename.toStdString().c_str(), std::ofstream::binary);
+  QByteArray ba(str.toStdString().c_str());
+  QByteArray bb = QByteArray::fromBase64(ba);
+  f.write(bb.constData(),bb.toStdString().size());
+  f.close();
+}
+
 void MainWindow::on_actionSa_ve_Scene_as_Script_triggered()
 {
+  if(scene->numberOfEntries() == 0)
+    return;
   QString filename =
       QFileDialog::getSaveFileName(this,
                                    "Save the Scene as a Script File",
                                    last_saved_dir,
                                    "Qt Script files (*.js)");
-  std::ofstream os(filename.toUtf8());
+  CGAL::Three::Three::CursorScopeGuard cs(Qt::WaitCursor);
+  std::ofstream os(filename.toUtf8(), std::ofstream::binary);
   if(!os)
     return;
-  std::vector<QString> names;
-  std::vector<QString> loaders;
+  std::vector<std::pair<QString, QString> > names;
+  std::vector<std::pair<QString, QString> > loaders;
   std::vector<QColor> colors;
   std::vector<int> rendering_modes;
   QStringList not_saved;
   for(int i = 0; i < scene->numberOfEntries(); ++i)
   {
     Scene_item* item = scene->item(i);
-    QString loader = item->property("loader_name").toString();
-    QString source = item->property("source filename").toString();
+    QString loader;// = item->property("loader_name").toString();
+    QString ext;
+    for(Polyhedron_demo_io_plugin_interface* iop : io_plugins)
+    {
+      if(iop->canSave(item))
+      {
+        QString sf = iop->saveNameFilters().split(";;").first();
+        //OFF Files (*.off)
+        QRegularExpression re("\\(\\*\\.(.*)\\)");
+        QRegularExpressionMatch rem = re.match(sf);
+        if(!rem.hasMatch())
+          continue;
+        ext = rem.captured(1);
+        QList<Scene_item*>to_save;
+        to_save.append(item);
+        QString savename(tr("%1.%2").arg(item->name()).arg(ext));
+        iop->save(QFileInfo(savename), to_save);
```

Comment:
If the current working directory is not writable, that will fail.

It should use [`QTemporaryFile`](https://doc.qt.io/qt-5/qtemporaryfile.html) within the temporary directory (`QDir::tempPath()`).